data-processor/src/main/java/io/micronaut/data/processor/mappers/jpa/TableAnnotationMapper.java

```diff
@@ -47,6 +48,8 @@ public String getName() {
         annotation.stringValue("name").ifPresent(builder::value);
         annotation.stringValue(SqlMembers.CATALOG).ifPresent(catalog -> builder.member(SqlMembers.CATALOG, catalog));
         annotation.stringValue(SqlMembers.SCHEMA).ifPresent(catalog -> builder.member(SqlMembers.SCHEMA, catalog));
+        Optional.ofNullable((AnnotationValue<Annotation>[]) annotation.getValues().get("indexes"))
```

Comment:
I would create an instance of `IndexAnnotationMapper` and use it to map the indexes.

Here you should map `indexes` of `@Table` to newly introduced `@Indexes` so that in the code, you only work with a new annotation.

Reply:
@dstepanov done

- [X]